packages/devextreme-angular/src/http/ajax.ts

```diff
@@ -0,0 +1,283 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import {
+  HttpClient, HttpEventType, HttpParams, HttpEvent, HttpErrorResponse, HttpResponse,
+} from '@angular/common/http';
+import { throwError, Subject } from 'rxjs';
+import { takeUntil, timeoutWith } from 'rxjs/operators';
+import { Deferred, DeferredObj } from 'devextreme/core/utils/deferred';
+import { isDefined } from 'devextreme/core/utils/type';
+import { getWindow } from 'devextreme/core/utils/window';
+import {
+  isCrossDomain,
+  evalCrossDomainScript,
+  getRequestOptions,
+  getJsonpCallbackName,
+  getRequestHeaders as getAjaxRequestHeaders,
+  getAcceptHeader,
+  getMethod,
+  evalScript,
+} from 'devextreme/core/utils/ajax_utils';
+
+type Result = Promise<any> & { abort: () => void };
+type DeferredResult = DeferredObj<any>;
+interface Options {
+  url: string;
+  [key: string]: any;
+}
+
+interface XHRSurrogate {
+  type?: string;
+  aborted: boolean;
+  abort: () => void;
+}
+
+const PARSER_ERROR = 'parsererror';
+const SUCCESS = 'success';
+const ERROR = 'error';
+const NO_CONTENT = 'nocontent';
+const TIMEOUT = 'timeout';
+const STATUS_ABORT = 0;
+const CONTENT_TYPE = 'Content-Type';
+const URLENCODED = 'application/x-www-form-urlencoded';
+
+function assignResponseProps(xhrSurrogate: XHRSurrogate, response: HttpResponse<any> | HttpErrorResponse) {
```

Comment:
I don't like the word surrogate for naming variable, maybe replace it with some synonim?

Reply:
Replacement(synonym of surrogate), proxy, mock

Reply:
I was thinking about this too